core/src/main/java/com/amplifyframework/analytics/AnalyticsCategoryBehavior.java

```diff
@@ -68,8 +70,10 @@
      * Examples of global properties would be `selectedPlan`, `campaignSource`
      *
      * @param properties Map of global properties wrapped as an instance of {@link Properties}
+     * @throws HubException thrown if an error condition is encountered while registering global properties
+     * and Amplify fails to publish the error message to the hub.
      */
-    void registerGlobalProperties(@NonNull Properties properties);
+    void registerGlobalProperties(@NonNull Properties properties) throws HubException;
```

Comment:
If you'll keep a checked exception on this API, then this should be `throws AnalyticsException` instead of `throws HubException`, since this is on an Analytics API.

If a `HubException` occurs in the implementation you should wrap it inside of an `AnalyticsException`, before returning it, here. Like:
```
try {
    Amplify.Hub.publish(....);
} catch (HubException publishFailure) {
    throw new AnalyticsException("Failed to register global properties." publishFailure, "Recovery suggestion TODO...");
}
```

The interface Javadoc could talk about how registration errors are communicated:
```
/**
 * ...
 * If registration fails, a (what kind?) of message is published via {@link Hub}.
 */
```